### PR TITLE
Towards BG-1126: Crypto checkout updates

### PR DESCRIPTION
**Pull request by ap-justin**
## Explanation of the solution
* update wiring as tx fees are no longer part of summary, (ultimately, fees are not part of checkout)  
---

* removed platform fees 
* show split values
* update wallet selection section
image

### As-builts : 
* Only connected wallet is shown (as our wallet api state is `ConnectedWallet | DisconnectedWallets[] | Loading`). Also retained a way for user to disconnect/change wallet
image

instead of 
image

* retain simulation states
image
image

* `Total donation` instead of `Donation to XXXXX` as there's no distinction yet between recipient and separate donation to better giving
image

* dollar values are async as it's not fetched from coingecko at the beginning of donation nor prefetched and set alongside tokens record in tokens table
(loading)
image
(loaded)
image

### Future
* https://linear.app/better-giving/issue/BG-1130/crypto-checkout-option-to-send-directly-to-donation-wallet
* https://linear.app/better-giving/issue/BG-1131/distinguish-between-bg-donation-and-endow-donation

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/components/donation/Steps/Submit/Crypto/TxSubmit/txPackage.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { EstimateResult, TxPackage } from "types/tx";
+import { ConnectedWallet, isCosmos, isEVM, isTerra } from "types/wallet";
+
+export const txPackage = (
```
Comment (ap-justin) [inline src/components/donation/Steps/Submit/Crypto/TxSubmit/txPackage.ts:4]: moved

File: src/components/donation/Steps/Submit/Crypto/TxSubmit/estimateDonation.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import { createTx } from "contracts/createTx/createTx";
+import { logger, scale, scaleToStr } from "helpers";
+import { estimateTx } from "helpers/tx";
+import { ChainID } from "types/chain";
+import { SimulContractTx, SimulSendNativeTx } from "types/evm";
+import { EstimateInput, TokenWithAmount } from "types/tx";
+import { EstimateStatus } from "../types";
+import { tokenBalance } from "./tokenBalance";
+
+export async function estimateDonation(
```
Comment (ap-justin) [inline src/components/donation/Steps/Submit/Crypto/TxSubmit/estimateDonation.ts:14]: moved and removed platform fee computation

File: src/components/donation/Steps/Submit/Crypto/Crypto.tsx [hunk 3]
```diff
@@ -1,128 +1,87 @@
-import { Info, LoadingStatus } from "components/Status";
+import { WalletProvider } from "@terra-money/wallet-provider";
```
Comment (ap-justin) [inline src/components/donation/Steps/Submit/Crypto/Crypto.tsx:1]: This component is previously `<Container/>` renamed to `Crypto` as it contains content which doesn't need `wallet`

File: src/components/donation/Steps/index.tsx [hunk 4]
```diff
@@ -14,11 +11,7 @@ export function Steps({ className = "", donaterConfig }: Props) {
     <div
       className={`grid ${className} w-full @container rounded overflow-clip bg-white min-h-96`}
     >
-      <WalletProvider {...chainOptions}>
```
Comment (ap-justin) [inline src/components/donation/Steps/index.tsx:17]: moved wallet packages down to crypto checkout

File: src/helpers/coin-gecko.ts [hunk 5]
```diff
@@ -1,15 +0,0 @@
-type Quotation = {
-  [index: string]: {
-    usd: number;
-  };
-};
-
-export const USD = async (coinId: string): Promise<number> =>
```
Comment (ap-justin) [inline src/helpers/coin-gecko.ts:7]: converted to RTKQ endpoint to cache usd rate result

File: src/components/donation/Steps/Submit/Crypto/Checkout.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import { maskAddress } from "helpers";
+import { PropsWithChildren } from "react";
+import { CryptoSubmitStep } from "slices/donation";
+import { ConnectedWallet } from "types/wallet";
+import TxSubmit from "./TxSubmit";
+import WalletSelection from "./WalletSelection";
+
+type Props = CryptoSubmitStep & { classes?: string };
+
+export default function Checkout({ classes = "", ...props }: Props) {
```
Comment (ap-justin) [inline src/components/donation/Steps/Submit/Crypto/Checkout.tsx:14]: this component is previously `<Crypto/>` but renamed to `<Checkout/>` as  gas fee (obtained via simulation which needs wallet) is now not included in donation summary items

**Review by SovereignAndrey — CHANGES_REQUESTED** [review]
This looks fantastic! Nice work! :heart_eyes: 

Only one tiny request around the change wallet button:
image
We should make the change wallet button match the other button styles:
- rounded corners
- have some padding around text label
- use title-case on label text (not uppercase)

File: src/components/donation/Steps/Submit/Crypto/Checkout.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  return (
+    <Container donation={props} wallet={wallet} classes={classes}>
+      <div className="flex items-center gap-2 mt-2">
+        <div className="size-2 rounded-full bg-green drop-shadow-[0_0_3px_rgba(126,198,130,0.9)]" />
+        <Image src={wallet.logo} className="size-5 rounded-full" />
+        <span className="text-sm">{maskAddress(wallet.address)}</span>
+        <button
+          type="button"
+          onClick={wallet.disconnect}
+          className="bg-gray-l5 hover:bg-gray-l4 text-xs px-2 py-1 rounded"
```
Comment (ap-justin) [inline src/components/donation/Steps/Submit/Crypto/Checkout.tsx:86]: @SovereignAndrey 
> We should make the change wallet button match the other button styles:

> rounded corners
have some padding around text label
use title-case on label text (not uppercase)

Applied requested changes
image

Comment (SovereignAndrey) [reply src/components/donation/Steps/Submit/Crypto/Checkout.tsx:86]: image
@ap-justin Change the button's class from `rounded` >> `rounded-full` I think it's OK.

Comment (ap-justin) [reply src/components/donation/Steps/Submit/Crypto/Checkout.tsx]: updated to rounded full https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2701/commits/1a3da1d4375c789c8071a4cdd2f54fde75f369b8
image

**Review by SovereignAndrey — APPROVED** [review]

**Pull request merged**